assets/app/scripts/directives/oscFileInput.js

```diff
@@ -31,6 +41,28 @@ angular.module('openshiftConsole')
           };
           reader.readAsText(file);
         });
+        $('.btn-file').on('click', function() {
+          delete scope.file;
+        });
+        $(dropArea).on('dragover', function(e) {
+          e.preventDefault();
+          e.stopPropagation();
+        });
+        $(dropArea).on('dragenter', function(e) {
+            e.preventDefault();
+            e.stopPropagation();
+        });
+        $(dropArea).on('drop', function(e) {
+          e.preventDefault();
+          e.stopPropagation();
+          if (e.originalEvent.dataTransfer){
+            if (e.originalEvent.dataTransfer.files.length > 0) {
+              scope.file = e.originalEvent.dataTransfer.files[0];
+              $(element).trigger('change');
```

Comment:
Is this to trigger the handler above? I'd rather pull that out into its own method and call it directly.